raster_analysis/results_store.py

```diff
@@ -65,75 +65,89 @@ def save_result(self, results: DataFrame, result_id: str) -> None:
                 RequestItems={TILED_RESULTS_TABLE_NAME: items}
             )
 
-        self.save_status(result_id, ResultStatus.success)
+        self.save_status(result_id, ResultStatus.success, str(i + 1))
 
     def save_status(
-        self, result_id: str, status: ResultStatus, detail: str = " "
+        self, result_id: str, status: ResultStatus, parts: int, detail: str = " ",
     ) -> None:
         dynamodb_client().put_item(
             TableName=TILED_STATUS_TABLE_NAME,
             Item={
-                "analysis_id": {"S": self.analysis_id},
                 "tile_id": {"S": result_id},
                 "status": {"S": status.value},
                 "detail": {"S": detail},
+                "parts": {"N": parts},
                 "time_to_live": {"N": self._get_ttl()},
             },
         )
 
-    def get_results(self) -> Dict[str, Any]:
-        results = dynamodb_client().query(
-            ExpressionAttributeValues={":id": {"S": self.analysis_id}},
-            KeyConditionExpression="analysis_id = :id",
-            TableName=TILED_RESULTS_TABLE_NAME,
+    def get_results(self, tiles: List[str]) -> Dict[str, Any]:
+        
+        # Fetching result parts for tile from status table to able to batch_get
+        # results that requires secondary index
+        result_statuses = self.get_statuses(tiles)
+        if len(result_statuses) == 0:
+            return []
+
+        tiles_and_result_parts = [
```

Comment:
Has anyone showed you how to run pre-commit yet? This will run automatic formatting and linting, which I think will make the formatting in the lines below a little neater. You can just run the command `pre-commit` from the root of the repo

Reply:
Definitely. I somehow missed that step when I pulled the repo.